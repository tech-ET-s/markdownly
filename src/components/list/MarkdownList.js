import React from 'react';
import PropTypes from 'prop-types';

function MarkdownList({ markdowns }) {
  const markdownList = markdowns.map((markdown, index) => (
    <li key={index}>
      <h1>{markdown.title}</h1>
    </li>
  ));
  return (
    <ul>
      {markdownList}
    </ul>
  );
}

MarkdownList.propTypes = {
  markdowns: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired
  })).isRequired
};

export default MarkdownList;
