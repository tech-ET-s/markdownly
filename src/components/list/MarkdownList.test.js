import React from 'react';
import { shallow } from 'enzyme';
import MarkdownList from './MarkdownList';

describe('MardownList component', () => {
  it('render mardown list', () => {
    const array = [{ title: 'calling' }, { title: 'bluf' }];
    const wrapper = shallow(<MarkdownList 
      markdowns={array}
      clickHandler={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
