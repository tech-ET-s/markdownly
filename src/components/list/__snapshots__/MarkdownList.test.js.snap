// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MardownList component render mardown list 1`] = `
<ul>
  <li
    key="0"
  >
    <h1>
      calling
    </h1>
  </li>
  <li
    key="1"
  >
    <h1>
      bluf
    </h1>
  </li>
</ul>
`;
